src/components/ChatEntry.js

```diff
@@ -2,21 +2,51 @@ import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
 
+function getDifferenceInYears(timeStamp) {
+  const today = new Date();
+  const year = today.getFullYear();
+
+  const pastDay = new Date(timeStamp);
+  const pastYear = pastDay.getFullYear();
+
+  const difference = year - pastYear;
+  return difference;
+}
+
 const ChatEntry = (props) => {
+  const timeInYears = getDifferenceInYears(props.timeStamp) + ' years ago';
+
+  const onLikeButtonClick = () => {
+    const updatedMessage = {
+        id: props.id,
+        sender: props.sender,
+        body: props.body,
+        timeStamp: props.timeStamp,
+        liked: !props.liked
+    };
+
+    props.onUpdateChatData(updatedMessage);
+  };
```

Comment:
If you refactor onUpdateChatData as I suggested in App.js, then your onLikeButtonClick() method only needs to modify the `liked` property of of a ChatEntry. If you follow this approach, you'd also need to introduce `useState` to keep track of a ChatEntry's like state. 

The benefit of only changing the `liked` property is that when a user clicks on a like, they really are only changing `liked`. Therefore, we shouldn't let the onLikeButtonClick event handler change id, sender, body, timestamp (which is more editing power than this method really needs). 

Reworking a solution so only `liked` is being affected will require you to do more than change a line or two of code, but it would best reflect what actually needs to happen. We should always prefer to only change the properties/data that need to be changed when we code.